File: Python/Year12/prep/decode-encode-askii.py
def convertStringToAscii(input_string: str) -> str:
    output = ""
    for char in input_string:
        output += str(ord(char)) + " "
    return output.strip()  # Strip trailing space


def convertAsciiToString(input_ascii: str) -> str:
    output = ""
    ascii_array = input_ascii.split(" ")
    for element in ascii_array:
        output += chr(int(element))
    return output

def main():
    while True:
        choice = int(input("1. Convert String To ASCII | 2. Convert ASCII to String | 3. Exit"))

        if choice == 1:
            input_string = input("Enter a string you want to convert to ASCII: ")
            ascii_result = convertStringToAscii(input_string)
            print("Converted to ASCII: ", ascii_result)
        if choice == 2:
            input_ascii = input("Enter ASCII you want to convert to a string: ")
            string_result = convertAsciiToString(input_ascii)
            print("Converted to string", string_result)
        if choice == 3:
            break
        else:
            print("Please enter a value between 1 and 3.")


if __name__ == '__main__':
    main()

File: Python/Year12/prep/decode-encode-askii.js
function convertStringToAscii(input) {
    var output = "";
    for (var i = 0; i < input.length; i++) {
        output += input.charCodeAt(i) + " ";
    }
    return output;
}
function convertAsciiToString(input) {
    var output = "";
    var asciiArray = input.split(" ");
    for (var _i = 0, asciiArray_1 = asciiArray; _i < asciiArray_1.length; _i++) {
        var element = asciiArray_1[_i];
        output += String.fromCharCode(parseInt(element));
    }
    return output;
}
var input = "Hello world";
var ascii = convertStringToAscii(input);
var ascii_2 = "72 101 108 108 111 32 119 111 114 108 100";
var string = convertAsciiToString(ascii_2);
console.log(ascii);
console.log(string);
